opendatahub-io/odh-dashboard
# Conditionally render delete modals

Towards [RHOAIENG-12117](https://issues.redhat.com/browse/RHOAIENG-12117)

## Description
Update usage of delete modals to render them only when they would be shown rather than having them be rendered but hidden.

## How Has This Been Tested?
Run thru the UI and check that delete modals are shown only when they should be shown.

- From the `Data science Projects` page.
  - use a kebab menu for a project and select `Delete project`
  - verify the delete modal is shown
- From a project details page
  - select the `Workbenches` tab
  - use a kebab menu for a project and select `Delete workbench`
  - verify the delete modal is shown
- From a project details page
  - select the `Pipelines` tab
  - use the top level kebab menu for a pipeline version and select `Delete pipeline server`
  - verify the delete modal is shown
- From a project details page
  - select the `Pipelines` tab
  - expand a pipeline that has pipeline versions
  - use a kebab menu for a pipeline version and select `Delete pipeline version`
  - verify the delete modal is shown
- From a project details page
  - select the `Pipelines` tab
  - expand a pipeline that has pipeline versions
  - and select a pipeline version
  - from the pipeline version page, use the actions menu and select `Delete pipeline version`
  - verify the delete modal is shown
- From a project details page
  - select the `Pipelines` tab
  - expand a pipeline that has pipeline versions
  - select a pipeline run
  - from the pipeline run page, use thee actions menu and select `Delete pipeline version`
  - verify the delete modal is shown
- From a project details page
  - select the `Pipelines` tab
  - for a that has no pipeline versions, use a kebab menu for the pipeline and select `Delete pipeline`
  - verify the delete modal is shown
- From a project details page
  - select the `Models` tab
  - from a models server in the list, use the kebab menu and select 'Delete model server'
  - verify the delete modal is shown.
- From a project details page
  - select the `Models` tab
  - expand the deployed models of a model server
  - from the deployed models list,  use the kebab menu and select 'Delete'
  - verify the delete modal is shown.
- From a project details page
  - select the `Cluster storage` tab
  - from a connection in the list, use the kebab menu and select 'Delete storage'
  - verify the delete modal is shown.
- From a project details page
  - select the `Connections` tab
  - from a connection in the list, use the kebab menu and select 'Delete'
  - verify the delete modal is shown.
- From a project details page
  - select the `Data connections` tab
  - from a data connection in the list, use the kebab menu and select 'Delete data connection'
  - verify the delete modal is shown.


## Test Impact
Current tests already test the delete modals.

## Request review criteria:
Self checklist (all need to be checked):
